File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    float cash;
    // do while to force correct input
    do
    {
        cash = get_float("Change owed: ");
    }
    while (cash < 0);

    //convert to cents
    int coins = round(cash * 100);
    //number of quarters
    int quarters = coins / 25;
    //remainder after quarters
    int quartersRm = coins - (25 * quarters);
    // number of dimes
    int dimes = quartersRm / 10;
    // remainder after dimes
    int dimesRm = quartersRm - (10 * dimes);
    // number of nickles
    int nickels = dimesRm / 5;
    //remainder after nickels
    int nickelsRm = dimesRm - (5 * nickels);
    // number of cents
    int cents = nickelsRm;
    //get the total
    int totalCoins = quarters + dimes + nickels + cents;
    printf("%d\n", totalCoins);

}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    # force valid value
    cash = get_float("Change owed: ")
    if cash >= 0:
        break
# List of coins
coins = [25, 10, 5, 1]
cents = cash * 100
totalcoins = 0
for coin in coins:
    coinsUsed = cents // coin
    # cents remaining
    cents -= coinsUsed * coin
    totalcoins += coinsUsed
    if cents < 1:
        break
print(totalcoins)
